block/block_test.go

```diff
@@ -49,3 +50,68 @@ func testBlocksRepository(t *testing.T, block_repo block.BlockRepository) {
 	// vacuum
 	assert.NoError(t, block_repo.Vacuum())
 }
+
+func testBlocksRepositoryIterator(t *testing.T, blocks_repo block.BlockRepository) {
+	negX, negY, negZ := block.NodeToBlock(-32000, -32000, -32000)
+	posX, posY, posZ := block.NodeToBlock(32000, 32000, 32000)
+
+	testData := []block.Block{
+		{negX, negY, negZ, []byte("negative")},
+		{0, 0, 0, []byte("zero")},
+		{posX, posY, posZ, []byte("positive")},
+	}
+	for i := range testData {
+		b := testData[i]
+		blocks_repo.Update(&b)
+	}
+	// logrus.SetLevel(logrus.DebugLevel)
+
+	// Helper function to loop over all channel data
+	// TODO(ronoaldo): simplify this so we don't have
+	// to work so verbose. Perhaps implement a wrapper
+	// to the channel?
+	consumeAll := func(it chan *block.Block) int {
+		t.Logf("consumeAll: fetching data from iterator")
+		count := 0
+		for {
+			select {
+			case b, ok := <-it:
+				if !ok {
+					return count
+				}
+				t.Logf("consumeAll: got %#v", b)
+				count++
+			case <-time.After(3 * time.Second):
+				t.Errorf("consumeAll: timed out")
+				return count
+			}
+			if count > 10 {
+				panic("consumeAll: too many items returned from channel")
+			}
+		}
+	}
+
+	// Fetch from neg -> pos, retrieves all three blocks
+	it, err := blocks_repo.Iterator(negX, negY, negZ)
+	if assert.NoError(t, err) {
```

Comment:
Check in the assertions the result order of the blocks, to ensure it is equally implemented across backends.